src/main/java/org/generationcp/middleware/dao/GermplasmListDataDAO.java

```diff
@@ -289,4 +291,25 @@ public GermplasmListData getByListIdAndGid(final Integer listId, final Integer g
 		return (result != null && result.size() > 0 ? (GermplasmListData) result.get(0) : null);
 
 	}
+
+	public void replaceGermplasm(final List<Integer> gidsToReplace, final Integer replaceWithGid) {
+		Preconditions.checkNotNull(gidsToReplace, "gidsToReplace passed cannot be null.");
+		Preconditions.checkArgument(!gidsToReplace.isEmpty(), "gidsToReplace passed cannot be empty.");
+		Preconditions.checkNotNull(replaceWithGid, "replaceWithGid passed in cannot be null.");
+
+		try {
+			final Query query =
+				this.getSession().createQuery(
+					"UPDATE GermplasmListData listData SET listData.gid = :replaceWithGid WHERE listData.gid in (:gidsToReplace)");
```

Comment:
We should also update `listdata.desig` (preferred name of selected germplasm) and l`istdata.grpname` (cross expansion of selected germplasm)